### DISCUSSION TITLE
Cannot kill actor, it was never added to the Scene

### DISCUSSION THREAD

**Discussion by robokop92** [excaliburjs/Excalibur#2509] [Q&A] [answered]
Hello ,
I've recently started using Excalibur for a project and ran in to the following problem:

When calling `actor.removeAllChildren()` some of the children remain and a warning in the console popus up `Cannot kill actor, it was never added to the Scene `

Here I've created an example of the problem https://codesandbox.io/s/brave-curie-i6di7r
You can see that after clicking the mouse a few times the number of "squares" increases to more than 4, because some of the actors still remain after calling `removeAllChildren()`

Am I misusing something here?

**Comment by chrisk-7777** [comment, marked as answer]
Out of interest, what version are you on @robokop92 ?
A similar bug cropped up semi-recently and was resolved (related discussion here: https://github.com/excaliburjs/Excalibur/issues/2419)

**Reply by chrisk-7777** [reply]
Whoops, should've checked out your sandbox! You're on 0.27, which doesn't include the fix discussed in #2419 

I typically work off the latest alpha (https://www.npmjs.com/package/excalibur) to get the latest and greatest. Obviously depends on your own workflow and if you're ok with alphas. Personally I haven't hit any major issues working this way with this project.

I'm not sure when 0.28 is being cut - @eonarheim would be able to field that one

**Reply by eonarheim** [reply]
I agree with Chris, @robokop92 the latest alpha should have a fix for this issue.

As for 0.28, I'm thinking early November. I do want to get a few bugs, Text, and Tiled improvements included before I cut the release.

**Reply by chrisk-7777** [reply]
Thanks Erik! Btw, I didn't want to put the pressure on regarding 0.28, I just couldn't answer to the details of the date

**Reply by robokop92** [reply]
Alright, thanks you for the answer!
